spec/unit/resource/rhsm_subscription_spec.rb

```diff
@@ -38,6 +48,35 @@
     expect { resource.action :remove }.not_to raise_error
   end
 
+  describe "#action_attach" do
+    context "when already attached to pool" do
+      before do
+        allow(provider).to receive(:subscription_attached?).with(resource.pool_id).and_return(true)
+      end
+
+      it "does not attach to pool" do
+        expect(provider).not_to receive(:shell_out!).with("subscription-manager attach --pool=#{resource.pool_id}")
+        resource.run_action(:attach)
+      end
+    end
+
+    context "when not attached to pool" do
+      before do
+        allow(provider).to receive(:subscription_attached?).with(resource.pool_id).and_return(false)
+      end
+
+      it "attaches to pool" do
+        expect(provider).to receive(:shell_out!).with("subscription-manager attach --pool=#{resource.pool_id}")
```

Comment:
@lamont-granquist I am unable to figure out why this isn't working properly. When I run the unit tests it's still trying to run `#shell_out!` on the actual host instead of rspec catching it and responding with the mock.